modules/widgets/src/splitter-widget.tsx

```diff
@@ -70,14 +79,62 @@ export class SplitterWidget extends WidgetImpl<SplitterWidgetProps> {
     render(
       <Splitter
         orientation={this.props.orientation}
-        initialSplit={this.props.initialSplit}
-        onChange={this.props.onChange}
+        initialSplitRatio={this.props.initialSplitRatio}
+        onChange={this.onSplitRatioChange}
         onDragStart={this.props.onDragStart}
         onDragEnd={this.props.onDragEnd}
       />,
       element
     );
   }
+
+  onSplitRatioChange = (splitRatio: number) => {
+    this.props.onSplitRatioChange(splitRatio);
+    this.splitRatio = splitRatio;
+    this._widgetManager!.viewsNeedUpdate = true;
+    this._widgetManager!.setNeedsRedraw('SplitterWidget');
+  };
+
+  filterViews(views: View[]): View[] {
```

Comment:
To contrast with the`layerFilter` API which is `Layer => boolean`, I wonder if this pattern is too powerful. If this function is implemented to add a view on every frame, will that make the number of view grow indefinitely?

Reply:
> wonder if this pattern is too powerful. If this function is implemented to add a view on every frame, will that make the number of view grow indefinitely?

- As you can see it replaces the existing views created by the app with clones, it does not add views. 
- If a widget wanted to add a view, it would do so with a specific id, and check for the presence of that id and only add if not already present.
- This setup does not add a view / filter views every frame. When the ratio changes, a flag is set that is cleared after the filtering is done.